portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,13 +38,40 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<String> commentMessages = new ArrayList<String>();
-    commentMessages.add("This website looks real stylish!");
-    commentMessages.add("I also am a lover of chicken alfredo!");
-    commentMessages.add("Wow! How long did this take you?");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment").addSort("timeOfComment", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> listOfComments = new ArrayList<Comment>();
+    // Creates a Comment object for each entity that was previously ever posted. 
+    for (Entity commentEntity : results.asIterable()) {
+      Date timeOfComment = (Date) commentEntity.getProperty("timeOfComment");
```

Comment:
Please make this its own function. In your Comment class, add static functions:

static Comment fromEntity(Entity) and 
static Entity toEntity(Comment)